Tests/BaseObjectTests.swift

```diff
@@ -16,181 +19,184 @@ class BaseObjectTests: QuickSpec {
             }
 
             beforeEach {
-                baseObject = BaseObject()
+                fake = FakeBaseObject()
                 callbackWasCalled = false
             }
 
-            describe("on") {
-                it("Callback should be called on event trigger") {
-                    baseObject.on(eventName, callback: callback)
-                    baseObject.trigger(eventName)
-
-                    expect(callbackWasCalled) == true
+            context("initialization") {
+                it("Should not initialize EventDispatcher twice") {
+                    expect(fake.eventDispatcher) == fake.eventDispatcher
                 }
+            }
 
-                it("Callback should receive userInfo on trigger with params") {
-                    var value = "Not Expected"
-                    baseObject.on(eventName) { userInfo in
-                        value = userInfo?["new_value"] as! String
-                    }
+            context("events") {
 
-                    baseObject.trigger(eventName, userInfo: ["new_value": "Expected"])
+                describe("on") {
+                    it("Callback should be called on event trigger") {
+                        fake.on(eventName, callback: callback)
+                        fake.trigger(eventName)
 
-                    expect(value) == "Expected"
-                }
+                        expect(callbackWasCalled) == true
+                    }
 
-                it("Callback should be called for every callback registered") {
-                    baseObject.on(eventName, callback: callback)
+                    it("Callback should receive userInfo on trigger with params") {
```

Comment:
Sometimes we forgot of the `it`. The `it` is part of the sentence, so `it callback should receive...` doesn't make much sense. We can rewrite like this: `it("receives userInfo when triggered with params")`.

We do not necessarily need to change this right now but keep it in mind for the next tests.